### DISCUSSION TITLE
Documentation question

### DISCUSSION THREAD

**Discussion by caeu** [R-nvim/R.nvim#265] [Q&A]
This is about the documentation under "R-built-in-terminal" section, which is not clear to me.
```
R.nvim sets the option "editor" to a function that makes the object to be
edited in a new tab when esc_term = true (the default value).
```
I prefer to use double `<esc><esc>` to switch to normal mode in terminal, hence I set `esc_term = false`
But not sure what is the side effect described in the above snippet from documentation.

**Comment by jalvesaq** [comment]
The option `esc_term` is from 2015 and I don't remember why I added this restriction. Can you make this simple change:

```diff
diff --git a/lua/r/run.lua b/lua/r/run.lua
index d05a13e..c952302 100644
--- a/lua/r/run.lua
+++ b/lua/r/run.lua
@@ -96,13 +96,7 @@ start_R2 = function()
     else
         table.insert(start_options, "options(nvimcom.nvimpager = TRUE)")
     end
-    if
-        type(config.external_term) == "boolean"
-        and not config.external_term
-        and config.esc_term
-    then
-        table.insert(start_options, "options(editor = nvimcom:::nvim.edit)")
-    end
+    table.insert(start_options, "options(editor = nvimcom:::nvim.edit)")
     if
         (type(config.external_term) == "boolean" and config.external_term == true)
         or type(config.external_term) == "string"
```

After this change, I have sent the following code to R:

```r
x <- c(1, 2, 3)
```

Then, on the R Console, I did:

```r
y <- edit(x)
```

With the above simple example, I didn't see any reason not to set the R's `editor` option when `esc_term = false`. It also works well when running RStudio as the `R_app`. So, perhaps, instead of trying to be too smart, we could create a new option (`set_R_editor`) with `true` as its default value.

**Comment by caeu** [comment]
Thanks, I tried it. Now I know there is an `edit()` function.

Running `y <- edit(x)` :

1) With the change:
Open a new buffer in another tab to edit the object.

2) Without the change:
Looks like it opens a nested instance of nvim inside the "nvim_R_terminal" to edit the object (see image).

Though in both cases, it works as expected. i.e. I can edit/save the new object.

<img width="731" alt="image" src="https://github.com/user-attachments/assets/8de4c5ac-b29a-4618-8e46-e2e5336b39a9">

**Comment by jalvesaq** [comment]
Thanks for the feedback! I'll wait about a week for comments from other people before adding the option `set_R_editor`.
